gateway/1.8/modules/ROOT/pages/flex-security-best-practices.adoc

```diff
@@ -0,0 +1,164 @@
+
+* Ensure Secrets are encrypted at rest and that least privilege access is enforced for the registration resource secret.
+* Store sensitive data, such as credentials and API keys, in secure vaults, for example, AWS Secrets Manager, Azure Key Vault, or HashiCorp Vault.
+* Avoid hardcoding secrets and only use environment variables for non-sensitive configurations.
+* Encrypt sensitive information in configuration files
+
+For Kubernetes deployments, see https://kubernetes.io/docs/concepts/security/secrets-good-practices/[Good practices for Kubernetes Secrets] for Kubernetes specific best practices.
+
+[[renew-the-flex-certificate]]
+== Renew the Flex Certificate 
+
+Anypoint Flex Gateway uses a managed PKI certificate to communicate with Anypoint Platform. This certificate is generated when you register Flex Gateway. Renew the certificate before it's expiration date to ensure the runtime can connect to the control plane.
+
+To learn how to renew your registration, see xref:flex-gateway-renew-certificate.adoc[].
+
+[[update-your-flex-gateway]]
+== Update your Flex Gateway
+
+MuleSoft regularly provides security updates for Flex Gateway. Keep your Flex Gateway updated to the latest version to ensure vulnerabilities are addressed. To reduce the risk of incompatibilities, follow safe change processes and upgrade sandbox runtimes ahead of production. 
+
+To learn more about how to update your Flex Gateway, see xref:flex-gateway-upgrade.adoc[].
+
+[[avoid-logging-payload-data]]
+== Avoid Logging Payload Data
+
+Runtime logs are available in the runtime layer through the `stdout` of the Flex Gateway pods and also in the Anypoint control plane for Connected Mode.
+
+To ensure client information is not leaked if runtime logs become compromised, avoid logging payload data in runtime logs. 
+
+If using other third party monitoring and analytics platforms, ensure you consider then in access reviews.
+
+[[secure-flex-gateway-service-pods]]
+== Secure Flex Gateway Service Pods
+
+Restrict access to Flex Gateway runtime pods to ensure unauthorized personel cannot access Flex Gateway configuration data or tamper with the runtime.
+
+For Kubernetes deployments, use Kubernetes Role-based access control (RBAC) to restrict access to execute commands in the pod. For more information, see https://kubernetes.io/docs/reference/access-authn-authz/rbac/[Using RBAC Authorization].
+
+In non-Kubernetes deployments, restrict access to the Flex Gateway virtual machine to open sessions.
+
+[[secure-redis-shared-storage]]
+== Secure Redis Shared Storage
+
+Flex Gateway caches both request data and runtime configurations (Connected Mode only) using Redis shared storage. An unauthorized user could obtain cached configuration data including Flex configuration or request processing caches or possibly disrupt the runtime. Secure the Redis service to ensure there is no unauthorized access.
```

Comment:
```suggestion
Flex Gateway uses Redis shared storage to cache request data and runtime configurations in Connected Mode only. An unauthorized user can obtain cached configuration data, such as Flex configuration or request processing caches, and can disrupt the runtime. You must secure the Redis service to prevent unauthorized access.
```